app/src/processing/app/Base.java

```diff
@@ -1804,10 +1804,9 @@ static public LibraryList getLibraries() {
   }
 
   public File getDefaultSketchbookFolderOrPromptForIt() {
-
     File sketchbookFolder = BaseNoGui.getDefaultSketchbookFolder();
 
-    if (sketchbookFolder == null) {
+    if (sketchbookFolder == null && !isCommandLine()) {
```

Comment:
As you noticed the folders handling code is spread around the IDE, and it needs a bit of love :-) Actually, I tried to factor all the folder handling code in a single class while doing this PR, but at some point I've just dropped this task to reduce the possibility of regressions.

The rationale behind this change is:
- if my calculations are correct `BaseNoGui.getDefaultSketchbookFolder()` will never return `null`
- this means that the `promptSketchbookLocation()` will never be called (and could be probably removed altogether)

but since I decided to not touch the folder handling functions in this PR, I opted to just leave it as it is and just add the "guard" to not invoke the file selector in cmd-line mode and let the IDE crash with a NPE (if that code path is ever taken in some obscure ways).